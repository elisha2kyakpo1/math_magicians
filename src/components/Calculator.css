.container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  margin-top: 20px;
  width: 500px;
  margin-left: auto;
  margin-right: auto;
}

.inner-div {
  grid-column: 1/ span 4;
}

.container > *:nth-child(18) {
  grid-column: 1 / span 2;
}

.input {
  border: none;
  background-color: #6a737d;
  color: #fff;
  width: 100%;
  height: 50px;
  font-size: 30px;
  text-align: right;
}

.o-buttons {
  background-color: #f66a0a;
  font-size: 50px;
  padding: 5px 40px;
}

button {
  background-color: #e1e4e8;
  border: 2px solid #d1d5da;
  padding: 5px 30px;
  font-size: 50px;
  cursor: pointer;
}
