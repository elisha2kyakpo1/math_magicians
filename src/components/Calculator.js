import React from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    this.setState((state) => calculate(state, e.target.textContent));
  }

  render() {
    const { next, total } = this.state;
    return (
      <div className="container">
        <div className="inner-div">
          <div className="input-div">
            <div className="input">{next || total || 0}</div>
          </div>

          <div className="buttons">
            <div className="numbers">
              <button type="button" onClick={this.handleClick} value="">AC</button>
              <button type="button" onClick={this.handleClick} value="">+/-</button>
              <button type="button" onClick={this.handleClick} value="">%</button>
              <button type="button" onClick={this.handleClick} value="">7</button>
              <button type="button" onClick={this.handleClick} value="">8</button>
              <button type="button" onClick={this.handleClick} value="">9</button>
              <button type="button" onClick={this.handleClick} value="">4</button>
              <button type="button" onClick={this.handleClick} value="">5</button>
              <button type="button" onClick={this.handleClick} value="">6</button>
              <button type="button" onClick={this.handleClick} value="">1</button>
              <button type="button" onClick={this.handleClick} value="">2</button>
              <button type="button" onClick={this.handleClick} value="">3</button>
              <button className="zero" type="button" onClick={this.handleClick} value="">0</button>
              <button type="button" onClick={this.handleClick} value="">.</button>
            </div>
            <div className="operators">
              <button type="button" onClick={this.handleClick} className="o-buttons" value="">รท</button>
              <button type="button" onClick={this.handleClick} className="o-buttons" value="">x</button>
              <button type="button" onClick={this.handleClick} className="o-buttons" value="">-</button>
              <button type="button" onClick={this.handleClick} className="o-buttons" value="">+</button>
              <button type="button" onClick={this.handleClick} className="o-buttons" value="">=</button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
