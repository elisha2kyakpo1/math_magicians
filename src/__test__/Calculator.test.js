import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('It should render the calculation result', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
