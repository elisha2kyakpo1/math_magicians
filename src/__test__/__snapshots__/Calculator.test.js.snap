// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the calculation result 1`] = `
<div
  className="container1"
>
  <h1
    className="math"
  >
    Let's do some math!
  </h1>
  <div
    className="container"
  >
    <div
      className="inner-div"
    >
      <div
        className="input"
      >
        0
      </div>
    </div>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="o-buttons"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="o-buttons"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="o-buttons"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="o-buttons"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="o-buttons"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
